test/test_scandir.py

```diff
@@ -45,6 +46,21 @@ def create_file(path, contents='1234'):
         f.write(contents)
 
 
+def create_utf8_name(length=None):
+    if length is None:
+        length = 15
+    else:
+        length = int(length)
+
+    utf8_chars = u'\uF10F\uD20D\uB30B\u9409\u8508\u5605\u3703\u1801'\
+                 u'\u0900\uF110\uD20E\uB30C\u940A\u8509\u5606\u3704'\
+                 u'\u1802\u0901\uF111\uD20F\uB30D\u940B\u850A\u5607'\
+                 u'\u3705\u1803\u0902\uF112\uD210\uB30E\u940C\u850B'\
+                 u'\u5608\u3706\u1804\u0903\u03A9\u2603'
+    return ''.join([random.choice(utf8_chars)
+                    for x in range(length)]).encode('utf-8')
```

Comment:
Your unit test looks *very* complex just to check the type of path type in scandir.walk(). Moreover, I don't see the "sys.platform != 'win32'" check in the unit test, so I expect that it will fail.